src/administration/api-keys.service.ts

```diff
@@ -146,3 +152,51 @@ export async function deleteApiKey({
 
   return createDeleteResponse(api_key_id, 'organization.project.api_key');
 }
+
+export async function listOrganizationApiKeys({
+  limit,
+  order,
+  order_by,
+  after,
+  before,
+  search
+}: OrganizationApiKeysListQuery) {
+  const filter: FilterQuery<Loaded<ProjectApiKey, 'project'>> = {};
+  const organizationUser = getOrganizationUser();
+
+  if (organizationUser.role !== OrganizationUserRole.OWNER) {
+    // get all project principals for organization user
+    const projectPrincipals = await ORM.em.getRepository(ProjectPrincipal).find(
+      {
+        principal: { type: PrincipalType.USER, user: ref(organizationUser) }
+      },
+      { filters: { projectAdministrationAccess: false } }
+    );
+
+    filter.$or = projectPrincipals.map((principal) =>
+      principal.role === ProjectRole.ADMIN
+        ? { project: principal.project }
+        : {
+            createdBy: principal.id,
+            project: principal.project
+          }
+    );
+  } else {
+    // get all project for organization user
+    const projects = await ORM.em.getRepository(Project).findAll();
+    // filter api-keys only by project
+    filter.project = { $in: projects.map((p) => p.id) };
+  }
```

Comment:
What's the purpose of this filter? It will always contain IDs of all projects.

Reply:
All projects in the current organization.